### DISCUSSION TITLE
How to copy sheet from one workbook to another

### DISCUSSION THREAD

**Discussion by prabhuvikas** [exceljs/exceljs#1707] [Q&A] [answered]
Hi,

Thanks for such a lovely library, i am enjoying it :)

I need to combine multiple `xlsx` workbook containing `1 sheet` each  in a `single workbook` containing `all sheets`.
For this what is the best way to copy sheets rather than looping through each row and cell.

**Comment by Siemienik** [comment, marked as answer]
Hi @prabhuvikas 

```js
const ws = /* ... */

const wsClone = newWb.addWorksheet('New')

wsClone.model = JSON.parse(JSON.stringify(ws.model));
```

This code should work perfectly:+1:

**Reply by prabhuvikas** [reply]
Thanks a lot :)

**Comment by CocafeDevBrew** [comment]
I am currently trying exactly the same but it completely loses the formatting. Like merged cells are not merged anymore and some text is weirdly bold, italic and strikethrough. Any suggestions?

[a comment by an automated account is not shown]
